app/models/user.rb

```diff
@@ -467,6 +467,7 @@ def banished?
 
   def subscribe_to_mailchimp_newsletter
     return unless registered
+    return if SiteConfig.mailchimp_api_key.blank? && SiteConfig.mailchimp_newsletter_id.blank?
     return unless email.present? && email.include?("@")
     return if saved_changes["unconfirmed_email"] && saved_changes["confirmation_sent_at"]
     return unless saved_changes.key?(:email) || saved_changes.key?(:email_newsletter)
```

Comment:
There are a lot of guard clauses here... I think it's still readable like this but not sure if we can make it more readable.